CMakeLists.txt

```diff
@@ -0,0 +1,88 @@
+# set range of known cmake versions where all policies are fine for us
+cmake_minimum_required(VERSION 3.1...3.13)
+if(${CMAKE_VERSION} VERSION_LESS 3.12) # do the best we can in this case
+    cmake_policy(VERSION ${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION})
+endif()
+
+file(STRINGS VERSION VERSION) # read hlwm version from file 'VERSION'
+project(Herbstluftwm
+	VERSION ${VERSION}
+	DESCRIPTION "A manual tiling window manager for X11")
+
+# write all executables to root of build directory
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
+
+## macros & functions (may be moved to cmake/ and included)
+# get project dependencies
+macro (get_external_deps) # – macro for scope
+	find_package(X11 REQUIRED)
+	include(FindPkgConfig)
+	pkg_check_modules(GLIB2 REQUIRED glib-2.0)
+endmacro()
+
+# extract git commit hash and append to project version
+function (set_gitversion)
```

Comment:
Why not put nicely encapsulated functions like this into some kind of utilities `.cmake` file?

Reply:
The comment above states that we could do it. However it would add a directory and a file and an include statement and in the end I didn't do it as right now we have only a couple of those.

Reply:
Sorry for not noticing the comment. Yeah, I'm fine keeping it that way. Maybe it'll work better with some more separation (like: not have this function figure out *and* set the version variable).

Reply:
Yes though, returning a value from a function? Too sophisticated for CMake! All you can do is pass another argument that tells the function which variable to set, and then have awkward code in the set statement. So yeah, that's that.